src/components/ActionModal.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+	<div class="overlay-action-modal" @click.self>
+		<div class="container-action-modal" @click.stop>
+			<h3>{{ title }}</h3>
+			<div v-if="isRename" class="rename">
+				<input id="rename-input" type="text" autocomplete="off">
+				<button class="cancel" @click="cancel">Cancel</button>
+				<button @click="rename">Save</button>
+			</div>
+			<div v-else>
+				<button class="cancel">Cancel</button>
+				<button>Save</button>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script lang="ts" setup>
+import { computed } from 'vue'
+const props = defineProps<{
+	type: 'rename' | 'delete'
+}>()
+const emits = defineEmits(['cancel', 'rename'])
+
+const isRename = props.type === 'rename'
+const title = computed(() => {
+	switch(props.type){
+		case 'delete':
+			return 'Are You Sure?'
+		case 'rename':
+			return 'Set New Name'
+		default:
+			throw new Error('Invalid modal type!')
+	}
+})
+
+function cancel(){
+	emits('cancel')
+}
+function rename(){
+	let inputValue = document.getElementById('rename-input').value
```

Comment:
You should probably use a `ref` to store inputValue and not manipulate DOM directly